### PR TITLE
Backport to Qt4.8

### PR DESCRIPTION
**Pull request by olegyurchenko**
For my old Qt4 project I make QtXlsxWriter simple and dirty Qt4 backport:
- Insert QStringLiteral macro
- Replace qt private classes QZipReader & QZipWriter to their copies QZlibReader & QZlibWriter
- Few not significant changes

### CODE REVIEW

File: src/xlsx/xlsxdocument.h [hunk 1]
```diff
@@ -34,7 +34,9 @@
 class QIODevice;
 class QImage;
 
-QT_BEGIN_NAMESPACE_XLSX
+//QT_BEGIN_NAMESPACE_XLSX
+namespace QXlsx {
```
Comment (VSRonin) [inline src/xlsx/xlsxdocument.h]: Why removing the macro?

File: src/xlsx/xlsxcellreference.cpp [hunk 2]
```diff
@@ -126,6 +131,14 @@ void CellReference::init(const QString &cell_str)
         _row = row_str.toInt();
         _column = col_from_name(col_str);
     }
+#else
+  if(cell_str.length() >= 2)
```
Comment (VSRonin) [inline src/xlsx/xlsxcellreference.cpp]: Use QRegExp instead

File: src/xlsx/qzlibwriter.h [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    void close();
+private:
+    QZLibWriterPrivate *d;
+    Q_DISABLE_COPY(QZLibWriter)
+};
+
+QT_END_NAMESPACE
+
+#endif // QT_NO_TEXTODFWRITER
+#endif // QZIPWRITER_H
```
Comment (VSRonin) [inline src/xlsx/qzlibwriter.h]: No need to import a file part of Qt sources

File: src/xlsx/qzlibreader.h [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    void close();
+
+private:
+    QZLibReaderPrivate *d;
+    Q_DISABLE_COPY(QZLibReader)
+};
+
+QT_END_NAMESPACE
+
+#endif // QZLIBREADER_H
```
Comment (VSRonin) [inline src/xlsx/qzlibreader.h]: No need to import a file part of Qt sources

File: src/xlsx/qzlib.cpp [hunk 5]
```diff
@@ -0,0 +1,1260 @@
...
+    writeUInt(eod.dir_start_offset, d->start_of_directory);
+    writeUShort(eod.comment_length, d->comment.length());
+
+    d->device->write((const char *)&eod, sizeof(EndOfDirectory));
+    d->device->write(d->comment);
+    d->device->close();
+}
+
+QT_END_NAMESPACE
+
```
Comment (VSRonin) [inline src/xlsx/qzlib.cpp]: No need to import a file part of Qt sources

File: src/xlsx/xlsxglobal.h [hunk 6]
```diff
@@ -26,6 +26,15 @@
 #define XLSXGLOBAL_H
 #include <QtGlobal>
 
+#if QT_VERSION < 0x050000
+#ifndef QStringLiteral
+# define QStringLiteral(str) QString::fromUtf8("" str "", sizeof(str) - 1)
```
Comment (VSRonin) [inline src/xlsx/xlsxglobal.h]: Why declare it if not used

File: src/xlsx/xlsxrelationships.cpp [hunk 7]
```diff
@@ -92,7 +92,7 @@ QList<XlsxRelationship> Relationships::relationships(const QString &type) const
 void Relationships::addRelationship(const QString &type, const QString &target, const QString &targetMode)
 {
     XlsxRelationship relation;
-    relation.id = QStringLiteral("rId%1").arg(m_relationships.size()+1);
+    relation.id = QString("rId%1").arg(m_relationships.size()+1);
```
Comment (VSRonin) [inline src/xlsx/xlsxrelationships.cpp]: Qt 5 should use QStringLiteral

File: src/xlsx/xlsxdocument.h (hunk 1 above)
Comment (olegyurchenko) [inline src/xlsx/xlsxdocument.h]: Qt4 moc problem

File: src/xlsx/qzlib.cpp (hunk 5 above)
Comment (olegyurchenko) [inline src/xlsx/qzlib.cpp]: * I am faced with the fact that in some builds of Qt, this module is absent:
oleg@noname:/usr/include/qt4$ qmake-qt4 -v
QMake version 2.01a
Using Qt version 4.8.6 in /usr/lib/x86_64-linux-gnu
oleg@noname:/usr/include/qt4$ find -name qzipreader_p.h
oleg@noname:/usr/include/qt4$
* But on other buids, it present (therefore, the name changed)
D:\Qt\4.8.6\include>qmake -v
QMake version 2.01a
Using Qt version 4.8.6 in D:\Qt\4.8.6\lib
D:\Qt\4.8.6\include>find . -name qzipreader_p.h
./QtGui/private/qzipreader_p.h

Comment (VSRonin) [inline src/xlsx/qzlib.cpp]: Private headers shouldn't ship in the includes as a general rule. You need to download the qt sources to get this header

**Pull request closed without merging**